From 6b3fa46d97201628b893f81c3bf3202aedd6ce30 Mon Sep 17 00:00:00 2001
From: Karel Zak <kzak@redhat.com>
Date: Thu, 27 Nov 2025 16:38:18 +0100
Subject: lsfd: fix const qualifier warning in strnrstr

Fix const qualifier discarded warning in strnrstr().
This warning is reported by gcc 15 which defaults to the C23 standard.

The function returns a non-const pointer into the haystack parameter,
and callers modify the string through that pointer. Therefore, the
haystack parameter should be char * rather than const char *.

Signed-off-by: Karel Zak <kzak@redhat.com>
(cherry picked from commit 935f2ab21add95059a92208f69ef578708307481)
---
 lsfd-cmd/file.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lsfd-cmd/file.c b/lsfd-cmd/file.c
index 32b08693d..5ea3ccc57 100644
--- a/lsfd-cmd/file.c
+++ b/lsfd-cmd/file.c
@@ -387,7 +387,7 @@ void decode_source(char *buf, size_t bufsize,
 			 dev_minor);
 }
 
-static char *strnrstr(const char *haystack, const char *needle, size_t needle_len)
+static char *strnrstr(char *haystack, const char *needle, size_t needle_len)
 {
 	char *last = strstr(haystack, needle);
 	if (last == NULL)
-- 
2.51.1

